README.md

```diff
@@ -1,9 +1,241 @@
+* [Contributing](#contributing)
+* [About](#about)
+* [License](#license)
+* [Change Log](#change-log)
+
+
+## Installation
+
+### Carthage
+
+[Carthage](https://github.com/Carthage/Carthage) is a dependency manager that builds your dependencies and provides you with binary frameworks that you can include in your project.
+
+You can install Carthage with Homebrew using the following command:
+
+```
+brew update
+brew install carthage
+```
+
+To download wolmo-core-iOS, add this to your Cartfile:
+
+```
+github "Wolox/wolmo-authentication-ios" "master"
+```
+//TODO: change for release version.
+
+### Manually
+[Bootstrap](#bootstrap) the project and then drag it to your workspace.
+
+
+## Bootstrap
+
+1. Clone the project.
+2. Run the bootstrap script that comes with it.
+
+With SSH: 
 
 ```
 git clone git@github.com:Wolox/wolmo-authentication-ios.git
+```
+
+Or with HTTPS:
+
+```
+git clone https://github.com/Wolox/wolmo-authentication-ios.git
+```
+
+And then:
+
+```
 cd wolmo-authentication-ios
 script/bootstrap
 ```
+
+
+
+## Dependencies
+
+- [WolmoCore](https://github.com/Wolox/wolmo-core-ios): Framework with useful basic extensions.
+- [ReactiveCocoa](https://github.com/ReactiveCocoa/ReactiveCocoa): Framework inspired by functional reactive programming.
+- [Rex](https://github.com/neilpa/Rex): Extensions for ReactiveCocoa.
+
+
+
+## Usage
+
+For more information about the framework's structure and concepts, see the [Framework Overview](FRAMEWORK-OVERVIEW.txt).
+
+There is an example application in this project called [AuthenticationDemo](AuthenticationDemo/AppDelegate.swift) in which the basic usage of the framework is shown.
+
+To use anything the framework offers, after installing it you just need to add `import Authentication` to your code, as you would do with any other library.
+
+The framework provides an `AuthenticationCoordinator` class that takes care of managing the authentication process and starting your app.
+
+So in your `AppDelegate.swift` file you will need to
+
+1. create the `AuthenticationCoordinator`
+2. trigger its `start` method
+
+and that's all !
+
+This will start the authentication process or redirect to your app if there is no need to authenticate.
+
+
+### How to create the `AuthenticationCoordinator`?
+
+ The `AuthenticationCoordinator` needs four things to be created:
+
+1. The UIWindow where the app will be shown.
+2. Which authentication screen you want to be shown first (login or signup).
+3. A session service with which the framework interacts for managing the user session.
+4. A component factory that provides the components for the framework to work with.
+
+
+
+#### Session Service
+
+You must provide a session service that comforms to the [SessionServiceType](Authentication/SessionServiceType.swift) protocol, which basically stores the current user and has endpoints for logging in and signing up with certain information (acquired by the framework from the final user).
```

Comment:
I don't like the term endpoints, as it resembles me to web. I would use 
... and provides functions for logging in and ...